src/Storages/ObjectStorage/StorageObjectStorageConfiguration.h

```
@@ -63,10 +81,21 @@ class StorageObjectStorageConfiguration
     /// buckets in S3. If object storage doesn't have any namepaces return empty string.
     virtual std::string getNamespaceType() const { return "namespace"; }
 
-    virtual Path getFullPath() const { return ""; }
-    virtual Path getPath() const = 0;
-    virtual void setPath(const Path & path) = 0;
 
+    // Path provided by the user in the query
+    virtual Path getRawPath() const = 0;
+    // Path used for reading, it is usually a globbed path like `'table_root/**.parquet'
+    Path getPathForRead() const;
```

> It is usually a globbed path like `'table_root/**.parquet'

In the default case it is not, so this comment is misleading.